subsys/bluetooth/mesh/pb_adv.c

```diff
@@ -445,16 +444,46 @@ static void gen_prov_start(struct prov_rx *rx, struct net_buf_simple *buf)
 		return;
 	}
 
-	prov_clear_tx();
-
-	link.rx.seg = (1 << (START_LAST_SEG(rx->gpc) + 1)) - 1;
 	link.rx.last_seg = START_LAST_SEG(rx->gpc);
+
+	if (link.rx.seg & BIT(0)) {
+		seg = ~link.rx.seg;
+		BT_WARN("Seg Start package losted, (seg: 0x%02x)", seg);
+
+		seg = find_msb_set(seg) - 1;
+		if (seg > link.rx.last_seg) {
+			BT_ERR("Invalid segment index %u", seg);
+			prov_failed(PROV_ERR_NVAL_FMT);
+			return;
+		}
+	}
+
+	if (link.rx.seg) {
+		seg = link.rx.seg;
+	}
+
+	link.rx.seg = (seg & (1 << (START_LAST_SEG(rx->gpc) + 1))) - 1;
```

Comment:
This doesn't look quite right. Shouldn't it be 
```C
link.rx.seg = seg & ((1 << (START_LAST_SEG(rx->gpc) + 1)) - 1);
```
as `seg` is already a bitmask?